### DISCUSSION TITLE
Dialog hideOnInteractOutside not propagating events

### DISCUSSION THREAD

**Discussion by georgekaran** [ariakit/ariakit#2089] [Questions] [answered]
Hi there, today we're facing a problem when dealing with the property `hideOnInteractOutside`, by customizing the callback to not close when clicking on an element that has the style `fixed`, the event is not being propagated to the element.

Our use case consists of dispatching some notifications above the `Modal` which the user cannot close due to events not being propagated.

Snippet:
```tsx
// Prevent closing the dialog when clicking on elements
// over the dialog (e.g.: notifications, tooltips, etc)
(event: Event): boolean => {
  const path = event.composedPath();

  return (
    path.find((element) => {
      if (element instanceof HTMLElement) {
        return window.getComputedStyle(element).position === "fixed";
      }

      return false;
    }) === undefined
  );
}
```

### Steps to reproduce the bug
1. Open sandbox: https://codesandbox.io/s/keen-bas-nv5xgw?file=/src/dialog.tsx:542-563
2. Open the modal
3. Click on the `Notify` button
4. Try to close the notification by clicking on the close button
5. The Modal is not being closed as expected but the event is not being propagated to the Notification button.

Do we have any workaround to fix this issue?

**Comment by diegohaz** [comment, marked as answer]
The Dialog behaves like that because of its `modal` state. Users can't interact with elements outside a modal dialog. `hideOnInteractOutside` only controls whether the Dialog will hide.

You can temporarily disable the `modal` state while showing the notification with something like this:

```jsx
const [modal, setModal] = useState(false)

<Dialog backdrop modal={modal} hideOnInteractOutside={...}>
  <Button
    onClick={() => {
      setModal(false)
      toast.success("Foo", { onClose: () => setModal(true) })
    }}
  />
</Dialog>
```

However, the notification element should ideally be considered a nested non-modal dialog while the original Dialog should keep its `modal` state. We need a better way to handle nested dialogs from third-party libraries.

**Reply by diegohaz** [reply]
Should be fixed by https://github.com/ariakit/ariakit/pull/2339
